source/rust_verify/src/lifetime_emit.rs

```diff
@@ -694,6 +694,11 @@ pub(crate) fn emit_fun_decl(state: &mut EmitState, f: &FunDecl) {
     state.push_indent();
     for (span, x, typ) in f.params.iter() {
         state.newline();
+        // For checking lifetimes, we make all function parameters mutable.
+        // This way, we do not have to closely track which parameters are
+        // mutable and which are not. This should not have an impact on
+        // the result of the lifetime analysis.
+        state.write("mut ");
```

Comment:
I'm not sure this is correct. `mut` checking is one of the things that is done by the "lifetime analysis" so I think we need to preserve this information from the source.

Reply:
can you elaborate on how `mut` is checked at this point? I thought that assigning to a non-mutable variable would have been caught at an earlier point.

Reply:
Well I don't know why rustc does it this way, but it does. mut-checking isn't done by this point, but it gets caught in the lifetime-checking stage instead.

Furthermore, Verus actually relies on the mut analysis being done correctly ... (there's actually an issue for it https://github.com/verus-lang/verus/issues/449) but even setting that issue aside, we still need to get the mut checks done in any case

Reply:
Oh, I see. I will address this. Thank you!

Reply:
done